Makefile

```diff
@@ -592,10 +592,9 @@ OCAML_LIBRARIES = $(COMPILERLIBS)
 $(foreach LIBRARY, $(OCAML_LIBRARIES),\
   $(eval $(call OCAML_LIBRARY,$(LIBRARY))))
 
-# Let the native libraries depend on the native compiler
+# Let the native (.cmx) files depend on the native compiler
 
-$(ocamlcommon_CMX_FILES) $(ocamlbytecomp_CMX_FILES) \
-$(ocamloptcomp_CMX_FILES) $(ocamltoplevel_CMX_FILES): ocamlopt$(EXE)
+%.cmx: ocamlopt$(EXE)
```

Comment:
I can never remember the rules - does this pattern rule definitely work for adding dependencies if the files also have specific entries in `.depend`? The previous version expressly listed the files.